### ISSUE TITLE
Recent changes to CPython's grammar and PEG parser, PEP-617

### ISSUE THREAD

**Issue by sobolevn** [RustPython/RustPython#2759]
## Feature
Python3.9+ support some new grammar rules, like `()` around `with` expression:

```pycon
» python3.9 --version
Python 3.9.4
            
» python3.9
Python 3.9.4 (default, Apr  5 2021, 01:47:16) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> with (open('a') as f):
...     print(f)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'a'
```

But, RustPython does not support it:
<img width="961" alt="Снимок экрана 2021-07-29 в 12 54 24" src="https://user-images.githubusercontent.com/4660275/127471814-00c35e98-ebeb-473e-a219-1a8d73814eba.png">

I think that the root of this problem is that new PEG parser is not supported: https://github.com/RustPython/RustPython/blob/master/parser/src/python.lalrpop

## Python Documentation

https://www.python.org/dev/peps/pep-0617/

**Comment by DimitrisJim** [comment]
Yup, the project is currently targeting 3.8 so this shouldn't come as a surprise. Hopefully, at some point in the future, 3.9 will be the new goal and using a PEG parser will probably be needed.

**Comment by sobolevn** [comment]
Then, I guess we can change this line in the README, because `3.9>=3.8` 🙂 
<img width="501" alt="Снимок экрана 2021-07-29 в 19 09 31" src="https://user-images.githubusercontent.com/4660275/127527041-337577e4-00eb-44d4-94ef-16e57d16390d.png">

**Comment by sobolevn** [comment]
Also: https://github.com/RustPython/RustPython/pull/2417

**Comment by youknowone** [comment]
I think we are practically not pinning a target. Any patch for 3.9+ features will be appreciated.

**Comment by fanninpm** [comment]
I believe 3.8 is the absolute minimum because it's referenced in `.github/workflows/ci.yaml`.

**Comment by youknowone** [comment]
we just can change it to 3.9 once we feel it is more convenient. that happened a few times.

**Comment by youknowone** [comment]
#2831

**Comment by sobolevn** [comment]
Related https://github.com/zsol/LibCST/blob/parser/native/libcst_parser/src/grammar.rs

**Comment by youknowone** [comment]
We'd better to rewrite our parser based on same parser definition with CPython.

**Comment by youknowone** [comment]
see also: https://github.com/charliermarsh/ruff/issues/286
